libs/python/helperApiTest.py

```diff
@@ -0,0 +1,178 @@
+import requests
+import json
+import logging
+import sys
+import os
+from os.path import exists
+
+if not os.path.exists("./log"):
+    os.mkdir("./log")
+
+file_handler = logging.FileHandler(filename='./log/crossconsumption.log', mode='w')
+stdout_handler = logging.StreamHandler(sys.stdout)
+handlers = [file_handler, stdout_handler]
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s:%(levelname)s:%(message)s',
+    handlers=handlers
+)
+
+def read_json_file(filepath):
+    try:
+        with open(filepath) as f:
+            return json.load(f)
+    except Exception as e:
+        logging.exception(e)
+
+def get_oauth_token(oauthurl, username, password):
+    client_id = username
+    client_secret = password
+    payload = {
+        "client_id": client_id, "client_secret": client_secret, "grant_type": "client_credentials"
+    }
+    headers = {
+        'Content-Type': 'application/x-www-form-urlencoded'
+    }
+    if oauthurl.endswith('/oauth/token') or oauthurl.endswith('/oauth/token?grant_type=client_credentials'):
+        url = oauthurl
+    elif oauthurl.endswith('auth/request'):
+        url = oauthurl
+        payload = json.dumps({
+            "apiKey": client_id, "secret": client_secret
+        })
+        headers = {
+            'Content-Type': 'application/json'
+        }
+    else:
+        url = f"{oauthurl}/oauth/token"
+    # payload = f"client_id={client_id}&client_secret={client_secret}&grant_type=client_credentials"
+    try:
+        response = requests.post(url, headers=headers, data=payload)
+        data = response.json()
+        response.raise_for_status()
+        if(response.status_code == 200):
+            if data.get("access_token", False):
+                access_token = data['access_token']
+                return access_token
+            else:
+                access_token = data['accessToken']
+                return access_token
+    except requests.exceptions.HTTPError as errh:
+        logging.exception(errh)
+    except requests.exceptions.RequestException as err:
+        logging.exception(err)
+
+def get_api_req(uri, access_token):
+    if not uri.startswith('https://'):
+        uri = 'https://' + uri
+    logging.info(f"The API request URL: {uri}")
+    headers = {
+        'Authorization': 'Bearer ' + access_token,
+    }
+    try:
+        response = requests.get(uri, headers=headers)
+        # response.json()
+        response.raise_for_status()
+        if response.status_code in [200, 201, 204]:
+            logging.info("GET request API call is successful!")
+    except requests.exceptions.HTTPError as errh:
+        logging.error("Failed the API request call")
+        logging.exception(errh)
+    except requests.exceptions.RequestException as err:
+        logging.exception(err)
+
+def get_api_req_basic(uri, username, password):
```

Comment:
There is quite some redundant code in the `get_api_req` function and in this method. Can this be merged in a more generic method to reduce duplicate code.